### PR TITLE
[weixin-app] fix api params

### PR DESCRIPTION
**Pull request by malash**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by jimexist — APPROVED** [review]

File: types/weixin-app/index.d.ts
Comment (taoqf) [inline types/weixin-app/index.d.ts]: Are you sure callback function is not needed here?

File: types/weixin-app/index.d.ts
Comment (taoqf) [inline types/weixin-app/index.d.ts]: I don't think `object` is a good type to use. maybe you could use `data:any`?

**Review by taoqf — CHANGES_REQUESTED** [review]

File: types/weixin-app/index.d.ts
Comment (malash) [reply types/weixin-app/index.d.ts]: As Mini Program documentation [wx.getStorage](https://developers.weixin.qq.com/miniprogram/dev/api/storage/wx.getStorage.html) response `data` should be `object` or `string` ( in fact it also could be `undefined`).

`any` means any type including `number` / `boolean` and so on and it's not fit to this situation.

File: types/weixin-app/index.d.ts
Comment (malash) [reply types/weixin-app/index.d.ts]: The type of `res` inside `OpenSettingOptions`'s `success` callback was wrong.

The wrong usage before this PR merged:

```ts
wx.openSetting({
  success(res) {
    // res.authSetting is AuthSetting
    // res.authSetting.success is function
  }
});
```

The correct usage:

```ts
wx.openSetting({
  success(res) {
    // res.authSetting is Scope
  }
});
```

Comment (taoqf) [reply types/weixin-app/index.d.ts]: Right.

Comment (malash) [reply types/weixin-app/index.d.ts]: Thanks. Could you please approve the PR if every thing OK ?

File: types/weixin-app/index.d.ts
Comment (taoqf) [reply types/weixin-app/index.d.ts]: You are right. but most time we would use `res.data.xxx` which will cause an error. I do not know the best way to resolve this.

Comment (malash) [reply types/weixin-app/index.d.ts]: I see. I changed `object` to `Record<string, any>` and is it OK ?

**Review by taoqf — APPROVED** [review]

**Pull request merged**
